### PR TITLE
refactor: use prisma types front-end-front-office

### PR DESCRIPTION
**Pull request by Apoorva64**
Ref: #253

### CODE REVIEW

**Comment by Apoorva64** [comment]
# ☁️ Nx Cloud Report

CI is running/has finished running commands for commit 9b3e0465187f5029d5995b9e2b15f619bdef45a5. As they complete they will appear below. Click to see the status, the terminal output, and the build insights.

[📂 See all runs for this branch](https://cloud.nx.app/branch?workspaceId=640ee8fb56da4d6500e89717&branchName=254,253-refactor-migrate-util-interface-types-quiz-type-to-prisma-types)

---


<details>
<summary>✅ Successfully ran 1 target </summary>

- [`nx affected --target=lint --parallel=3`](https://cloud.nx.app/runs/6ckfZQvxst)
</details>

---

Sent with 💌 from [NxCloud.](https://nx.app)

**Review by Guekka — COMMENTED** [review]
How does `Prisma.QuestioNGetPayload` work?
We also might want to add a function, you have duplicated code

**Comment by Apoorva64** [comment]
> How does `Prisma.QuestioNGetPayload` work?
> We also might want to add a function, you have duplicated code

Rtfm xD
https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types

**Comment by Guekka** [comment]
> Rtfm 

Alright, I deserve it

**Review by Guekka — CHANGES_REQUESTED** [review]
Refactor out this bit
```ts
Prisma.QuestionGetPayload<{
    include: {
      answers: true;
      clues: true;
    };
  }>;
```

**Review by ArnaudDumanois — DISMISSED** [review]
lgtm

**Comment by Guekka** [comment]
Looks like there's a CI failure. Working on it

File: apps/front-end/back-office/src/app/components/quiz-creation/question-detail-answers/question-detail-answers.component.ts [hunk 1]
```diff
@@ -7,11 +8,16 @@ import { Quiz } from '@webonjour/util-interface';
   styleUrls: ['./question-detail-answers.component.scss'],
 })
 export class QuestionDetailAnswersComponent {
-  @Input() question!: Quiz.Question;
+  @Input()
+  question!: Prisma.QuestionGetPayload<Quiz.QuestionWithAnswersAndClues>;
 
   onAddAnswer() {
     this.question.answers.push({
+      id: -1,
```
Comment (Apoorva64) [inline apps/front-end/back-office/src/app/components/quiz-creation/question-detail-answers/question-detail-answers.component.ts:16]: Should use Input type

**Review by ArnaudDumanois — DISMISSED** [review]
lgtm

File: apps/front-end/back-office/src/app/components/quiz-creation/question-detail-main/question-detail-main.component.spec.ts [hunk 2]
```diff
@@ -15,11 +15,13 @@ describe('QuestionDetailMainComponent', () => {
...
     component.question = {
-      id: '',
-      answers: [],
-      clues: [],
       title: 'test',
-      type: Quiz.QuestionType.CHOICE,
+      type: QuestionType.CHOICE,
+      quiz: {
+        connect: {
+          id: 1,
```
Comment (ozeliurs) [inline apps/front-end/back-office/src/app/components/quiz-creation/question-detail-main/question-detail-main.component.spec.ts:22]: wtf dude

File: libs/data-access-fake-backend/src/lib/interceptors/fake-backend.interceptor.ts [hunk 3]
```diff
@@ -198,15 +202,15 @@ export class FakeBackendInterceptor implements HttpInterceptor {
...
+      const quizzes = quizList.filter((x) => quizIds.includes(String(x.id)));
       return ok(quizzes);
     }
 
     function getAllAccommodation() {
       return ok(accommodationMocks);
     }
 
-    function deleteQuiz(id: string) {
+    function deleteQuiz(id: number | undefined) {
```
Comment (ozeliurs) [inline libs/data-access-fake-backend/src/lib/interceptors/fake-backend.interceptor.ts:213]: cannot delete undefined quiz

File: libs/data-access-fake-backend/src/lib/interceptors/fake-backend.interceptor.ts [hunk 4]
```diff
@@ -16,12 +16,13 @@ import {
   of,
   throwError,
 } from 'rxjs';
-import { Auth, Patient, Quiz } from '@webonjour/util-interface';
+import { Auth, Patient, Quiz as Q } from '@webonjour/util-interface';
```
Comment (ozeliurs) [inline libs/data-access-fake-backend/src/lib/interceptors/fake-backend.interceptor.ts:19]: Do not use single letter variable

File: apps/front-end/back-office/src/app/components/quiz-creation/question-detail-answers/question-detail-answers.component.ts (hunk 1 above)
Comment (Guekka) [reply apps/front-end/back-office/src/app/components/quiz-creation/question-detail-answers/question-detail-answers.component.ts:16]: Will do in another issue

File: apps/front-end/back-office/src/app/components/quiz-edition/quiz-edition.component.ts [hunk 5]
```diff
@@ -4,20 +4,23 @@ import { Quiz } from '@webonjour/util-interface';
...
+type Question = Prisma.QuestionGetPayload<Quiz.QuestionWithAnswersAndClues>;
 
 @Component({
   selector: 'webonjour-quiz-edition',
   templateUrl: './quiz-edition.component.html',
   styleUrls: ['./quiz-edition.component.scss'],
 })
 export class QuizEditionComponent implements OnInit, AfterViewInit {
-  quiz!: Quiz.Quiz;
+  quiz!: Prisma.QuizGetPayload<Quiz.QuizWithQuestions>;
```
Comment (Guekka) [inline apps/front-end/back-office/src/app/components/quiz-edition/quiz-edition.component.ts:17]: Use the above `Question` type

**Review by Guekka — APPROVED** [review]

**Pull request merged**